tools/xnvme.c

```diff
@@ -897,6 +925,22 @@ static struct xnvmec_sub g_subs[] = {
 			{0},
 		},
 	},
+	{
+		"dsm",
+		"Dataset Management",
+		"Dataset Management",
+		sub_dsm,
+		{
+			{XNVMEC_OPT_URI, XNVMEC_POSA},
+			{XNVMEC_OPT_NSID, XNVMEC_LREQ},
+			{XNVMEC_OPT_NR, XNVMEC_LREQ},
+			{XNVMEC_OPT_AD, XNVMEC_LREQ},
+			{XNVMEC_OPT_IDW, XNVMEC_LREQ},
+			{XNVMEC_OPT_IDR, XNVMEC_LREQ},
+			{XNVMEC_OPT_DEV_NSID, XNVMEC_LOPT},
+			{XNVMEC_OPT_BE, XNVMEC_LOPT},
```

Comment:
For the user to supply the range, it could use SLBA and NLB here.
Should be possible to supply context-attributes as well.

Reply:
Done